.github/actions/run-sql-script/action.yml

```diff
@@ -0,0 +1,49 @@
+name: 'run-sql-script'
+description: 'Run a script against a CloudSQL instance'
+author: 'dd'
+inputs:
+  db-connection-name:
+    description: 'full CloudSQL connection name: <project>:<region>:<cloudsql-instance>'
+    required: true
+  db-name:
+    description: 'name of the database to connect to'
+    required: true
+  db-password:
+    description: 'password to login to the postgres database'
+    required: true
+  db-username:
+    description: 'username to login to the postgres database'
+    required: true
+  cloudsql-sa-file:
+    description: 'path to the CloudSQL SA json file somewhere'
+    required: true
+  script-file:
+    description: 'path to the SQL script file'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+  - name: run script on db
+    shell: bash
+    run: |
+      safile="${{ inputs.cloudsql-sa-file }}"
+      creddir=$(cd "${safile%/*}" && pwd)
+      credfile="${safile##*/}"
+      # We can parameterize this if need be, but this should work
+      port=5434
+      echo "Launching the CloudSQL proxy"
+      dod=$(docker run --rm -d \
+        -v ${creddir}:/config \
+        -p 127.0.0.1:${port}:${port} \
```

Comment:
I'd make host ip and port parameters with these values as defaults. The  IP is in at least two places.